lib/net/http/header.rb

```diff
@@ -806,6 +806,107 @@ def set_form_data(params, sep = '&')
 
   alias form_data= set_form_data
 
+  # Stores form data to be included in the body of a +POST+ or +PUT+ request.
+  # When the request is sent, the data is URL-encoded
+  # (see {URI#encode_www_form}[https://docs.ruby-lang.org/en/master/URI.html#method-c-encode_www_form])
```

Comment:
This is only true if `enctype` is `application/x-www-form-urlencoded`. If `enctype` is `multipart/form-data`, then it's encoded differently. For example, here I'll directly use the private method `encode_multipart_form_data` to encode a hash as `multipart/form-data`:


```ruby
uri = URI('http://example.com')
req = Net::HTTP::Get.new(uri)
buf = StringIO.new
req.send(:encode_multipart_form_data, buf, {"testing" => "hello"}, {})
puts buf.string
# --VW1nIsu7UHRJGW-ROHq6aeAadUQ87H4CsUoCvcyYSBK6qxQu43nnVg
# Content-Disposition: form-data; name="testing"
#
# hello
# --VW1nIsu7UHRJGW-ROHq6aeAadUQ87H4CsUoCvcyYSBK6qxQu43nnVg--
```

Reply:
Removed.